client/src/components/SocialShare.tsx

```diff
@@ -98,9 +100,11 @@ const SocialSharePortfolioWithoutI18n: React.FC<{
       i18n={i18n}
       location={location}
       url={encodeURI(
-        `https://whoownswhat.justfix.nyc/address/${addr.boro}/${addr.housenumber}/${
-          addr.streetname
-        }${location === "summary-tab" ? "/summary" : ""}`
+        `https://whoownswhat.justfix.nyc${createRouteForAddressPage(
+          addr.boro,
+          addr.streetname,
+          addr.housenumber
+        )}${location === "summary-tab" ? "/summary" : ""}`
       ).replace(" ", "%20")} // Support for Android
```

Comment:
Hmm I wonder if we should also factor out an `absoluteURL()` function that makes a relative route into an absolute one?  That way we also wouldn't have to hard-code the URL to WoW here... Also we should probably generate this URL's protocol/host from `window.location` so that it works regardless of the domain it's hosted on. I guess we could also do that in a separate PR if you want though.  But it seems useful since there are a bunch of other places where `whoownswhat.justfix.nyc` is hard-coded in this PR (obviously it's not your fault since it's from legacy code!).

Err also, I think you probably want to be using `encodeURIComponent()` for the street name and/or house number in  `createRouteForAddressPage`, rather than `encodeURI()` on the whole thing--that will _probably_ fix the "support for android" workaround you have in there, and even if it doesn't, I think that workaround should be done in `createRouteForAddressPage`, since that's really the source of any space characters that appear in the URI... Otherwise you'll have to do that workaround everywhere you use `createRouteForAddressPage()` which isn't very DRY!

Reply:
All set! I went ahead and factored out all `encodeURI()` instances while we're at it